### PR TITLE
providers: rename ec2 -> aws, gce -> gcp in non-legacy mode

### PR DESCRIPTION
**Pull request by bgilbert**
https://github.com/coreos/fedora-coreos-tracker/issues/162

Requires https://github.com/coreos/coreos-metadata/pull/181.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,95 @@ This can include adding SSH keys and writing cloud-specific attributes into an e
...
+
+  - custom
+    - Attributes
+      - COREOS_CUSTOM_HOSTNAME
+      - COREOS_CUSTOM_PUBLIC_IPV4
+      - COREOS_CUSTOM_PRIVATE_IPV4
+      - COREOS_CUSTOM_PUBLIC_IPV6
+      - COREOS_CUSTOM_PRIVATE_IPV6
+
+### Container Linux Legacy Support
```
Comment (lucab) [inline README.md]: I'd suggest we move this to its own `cl-legacy.md` doc (and update any links under `coreos/docs` repo) to keep the README to a sane amount of information load.

File: src/metadata.rs [hunk 2]
```diff
@@ -34,16 +34,40 @@ macro_rules! box_result {
...
+        }
         "azure" => box_result!(Azure::try_new()?),
         "cloudstack-metadata" => box_result!(CloudstackNetwork::try_new()?),
         "cloudstack-configdrive" => box_result!(ConfigDrive::try_new()?),
         "digitalocean" => box_result!(DigitalOceanProvider::try_new()?),
-        "ec2" => box_result!(Ec2Provider::try_new()?),
-        "gce" => box_result!(GceProvider::try_new()?),
+        // FIXME: "ec2" is accepted in non-legacy mode temporarily for transition
+        "ec2" => box_result!(AwsProvider::try_new()?),
+        "gce" => {
```
Comment (lucab) [inline src/metadata.rs]: Question: doesn't a `#[cfg(feature = "cl-legacy")]` on the match-arm work? (same for all branches)

**Review by lucab — COMMENTED** [review]
No major things, just a style question and a doc-organization nit.

**Comment by bgilbert** [comment]
Updated.

File: README.md (hunk 1 above)
Comment (bgilbert) [reply README.md]: https://github.com/coreos/docs/pull/1272

**Review by lucab — APPROVED** [review]
LGTM!

**Pull request merged**